### ISSUE TITLE
A button for Light and Dark mode

### ISSUE THREAD

**Issue by Skrillx13** [tomfran/typo#27]
It would be nice to have a button that allows you to actually switch between light and dark mode.

**Comment by tomfran** [comment]
Yes I agree, need to find the right icon tho. I'll think of something

**Comment by Skrillx13** [comment]
> Yes I agree, need to find the right icon tho. I'll think of something

Might I suggest using these light and dark icons from [Material Design](https://pictogrammers.com/library/mdi/)? I've used them before and they are quite nice.

**Comment by neel04** [comment]
@tomfran You can check the icons on my website (https://neel04.github.io/my-website/) I've done a few changes to the colorscheme, but it'll fit right in to `typo` too I believe

**Comment by otaliptus** [comment]
Hey, I can take this one. 

Only question is how should I configure the logic in the config? There's `theme` and `colorPalette`, and my intuition would be adding a ```colorPalette2``` to the config and showing the mode button on the top right in case two palettes are different, to not to add more fields to the toml file.

Open for better ideas, did not think deep on it. 

Here's a suggestion for the icons.

![image](https://github.com/user-attachments/assets/86e833f3-a88f-4efe-9be7-44273f1c9a17)

**Comment by Skrillx13** [comment]
> Hey, I can take this one.
> 
> Only question is how should I configure the logic in the config? There's `theme` and `colorPalette`, and my intuition would be adding a `colorPalette2` to the config and showing the mode button on the top right in case two palettes are different, to not to add more fields to the toml file.
> 
> Open for better ideas, did not think deep on it.
> 
> Here's a suggestion for the icons (let me know if you're looking for something much more aesthetic, this is just simple svg).
> 
> ![image](https://private-user-images.githubusercontent.com/56600661/379740585-86e833f3-a88f-4efe-9be7-44273f1c9a17.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk3Njg5ODEsIm5iZiI6MTcyOTc2ODY4MSwicGF0aCI6Ii81NjYwMDY2MS8zNzk3NDA1ODUtODZlODMzZjMtYTg4Zi00ZWZlLTliZTctNDQyNzNmMWM5YTE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI0VDExMTgwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA3ZGQ2YTY0YmVmYWQwODA5NzNiZGU4YjVjNTVlZGM2MGEzMGVjYTcyNTc3MGRmMjllNzQwZDBhN2Y4ZDhmZDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vnhDoaP20gF-afXkBnCDy0zGYEwcKir4alumF26t0gc)

I think ColorPallete 2 would be ok

**Comment by tomfran** [comment]
> Hey, I can take this one. 
> 
> 
> 
> Only question is how should I configure the logic in the config? There's `theme` and `colorPalette`, and my intuition would be adding a ```colorPalette2``` to the config and showing the mode button on the top right in case two palettes are different, to not to add more fields to the toml file.
> 
> 
> 
> Open for better ideas, did not think deep on it. 
> 
> 
> 
> Here's a suggestion for the icons (let me know if you're looking for something much more aesthetic, this is just simple svg).
> 
> 
> 
> ![image](https://github.com/user-attachments/assets/86e833f3-a88f-4efe-9be7-44273f1c9a17)
> 
> 

Hello, sorry but why are you thinking of a color palette 2? 
My idea would be to introduce a new "manual" theme mode, alongside the existing auto, light and dark. Perhaps we could specify what theme to use on the first visit, either dark or light.

The color palette stuff is independent, having two different palettes, one for light and one for dark wasn't contemplated.

**Comment by Skrillx13** [comment]
> > Hey, I can take this one.
> > Only question is how should I configure the logic in the config? There's `theme` and `colorPalette`, and my intuition would be adding a `colorPalette2` to the config and showing the mode button on the top right in case two palettes are different, to not to add more fields to the toml file.
> > Open for better ideas, did not think deep on it.
> 
> Hello @otaliptus and @Skrillx13, perhaps I didn't get your idea, but why are you thinking of a color palette 2? My idea would be to introduce a new "manual" theme mode, alongside the existing auto, light and dark. Perhaps we could specify what theme to use on the first visit, either dark or light.
> 
> The color palette stuff is independent, having two different palettes, one for light and one for dark respectively, wasn't contemplated. Also I feel this would complicate configuration, additionally, one can currently achieve this by defining a custom palette.
> 
> About the icons, yes those are really generic ones, but I guess we can always change them in the future/add a configuration for them.

I suppose that would work better. As for the icons, configuration could be added to allow the user to choose an icon from like an icon library.

**Comment by RomanAtBrief** [comment]
Hi @tomfran have you managed to add a toggle for light/dark mode?
Or maybe you have a workaround for now?
Thank you

**Comment by tomfran** [comment]
Hi, haven't added it, perhaps [hooks](https://tomfran.github.io/typo-wiki/features/advanced-customization/#hooks) can be a workaround.
